### ISSUE TITLE
URL of multiline link gets cut off

### ISSUE THREAD

**Issue by edu-baer** [typst/typst#3478]
### Description

During a lecture, a couple of my students informed me about a link in an exercise PDF (written with Typst) was "not working".
The URL of the link was cut off at the line break and therefore the website was not found.

![image](https://github.com/typst/typst/assets/122176859/b00335af-81f5-474f-a6d1-f3f9671f1d5f)

I later found out it is dependent on the cursor position. If the cursor is right on top of the text, the URL is correct. If the cursor is slightly above or below the first line, the URL is cut off. Notice the change of the cursor!

![output](https://github.com/typst/typst/assets/122176859/72cfb31b-df2b-4bb5-aa1d-56b0c6fbb749)

This behavior only occurs if there is some text in front of the link.

![link_issue_no_text](https://github.com/typst/typst/assets/122176859/83c7b681-13db-468a-bc66-a96633fd826b)

If there is no space between the text and the link, the text gets prepended to the URL. However, this does not work with all text ("a" is prepended but "text" is not). 

![link_issue_no_space](https://github.com/typst/typst/assets/122176859/5b2228ba-52c8-489f-9e02-fcfb01baadea)

All this time, I was using the Link function with the URL as the only parameter. For example:
```
text #link("https://www-genesis.destatis.de/genesis/online?sequenz=tabelleErgebnis&selectionname=61111-0001&startjahr=2012&endjahr=2022")
```

I then remembered that you can just write the link without the link function.
```
text https://www-genesis.destatis.de/genesis/online?sequenz=tabelleErgebnis&selectionname=61111-0001&startjahr=2012&endjahr=2022
```

This leads to the same behavior.

Supplying content to the link function also showed some strange behavior. While some random text works as expected, a URL as content overwrites the URL of the link.

![link_issue_with_content](https://github.com/typst/typst/assets/122176859/d749abe0-fe5e-4776-97f8-9aed8954b1a1)

I am not sure if this is intended behavior or if it is even related to the other issue.

Long story short, I have no idea what exactly causes this, but I am pretty sure it's wrong. Maybe the text gets reinterpreted as a URL if it matches some sort of regex? I don't know.

The PDF was compiled using Typst version 0.10.0. I tested it in Google Chrome and Adobe Acrobat Reader and they both behave in the same way. The screenshots and GIFs are of Adobe Acrobat Reader.

The sources I used for testing can be viewed using the reproduction URL.

### Reproduction URL

https://typst.app/project/rloJEjhiCFkNQYO4nWW3Rs

### Typst version

- [X] I am using the latest version of Typst

**Comment by istudyatuni** [comment]
What PDF editor are you using? Cannot reproduce in Okular

**Comment by EpicEricEE** [comment]
Seems to be a duplicate of #2587

**Comment by edu-baer** [comment]
@EpicEricEE you are correct. I am a bit annoyed that I did not find it while searching.

Adobe Acrobat Reader seems to do something similar to Chromium. It also generates a link with a larger clickable area than what is specified in the PDF.

The document I was referring to (the exercise) was originally written with LaTeX. The line break does not affect the URL in that version using Acrobat.

You can also notice the cursor change, presumably switching from an Acrobat-generated URL to the link URL. However, the URL stays the same.

![image](https://github.com/typst/typst/assets/122176859/cccd190d-b7fe-4ebd-91ac-699316342f62)

![image](https://github.com/typst/typst/assets/122176859/323b3e74-4b21-4641-bf98-5462a71f28d9)

I also tested your LaTeX example, but Acrobat does not generate a cut-off link in that case.

Maybe there is something Typst can do to improve links.

**Comment by laurmaedje** [comment]
I'll close this as duplicate and can't-fix. If somebody does find a way to somehow improve this from our side, please point it out!
